src/iiif/views.py

```diff
@@ -30,17 +30,17 @@ def index(request, iiif_url):
 
 # TODO: limit to dataportaal urls
 @csrf_exempt
-@ratelimit(key='ip', rate='3/d')  # TODO: Check django cache settings for rate limiter to work across paralel docker containers
+@ratelimit(key='ip', rate='3/d')  # TODO: Check django cache settings for rate limiter to work across parallel docker containers
 def send_dataportaal_login_url_to_burger_email_address(request):
     try:
         # Some basic sanity checks
         tools.check_for_post(request)
         payload = tools.parse_payload(request)
-        tools.check_login_url_payload(payload)
-        tools.check_email_validity(payload['email'])
+        email, origin_url = tools.check_login_url_payload(payload)
+        tools.check_email_validity(email)
```

Comment:
There is way to much business logic inside tools.py. Why not use Django rest framework to handle a lot of boilerplate stuff (e.g. check_for_post() kind of stuff)